docs: known issue for listener using `proxy_protocol _behavior` (#27628)
hashicorp/vault: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx

<a id="listener-proxy-protocol-config" />

### Potential DoS when using the `deny_unauthorized` proxy protocol behavior for a TCP listener

#### Affected versions

##### Community Edition (CE)

- 1.10.x - 1.15.x

> - 1.17.1

Reviewer: Should this be 1.17.0 and 1.17.1?

Author: No only `1.17.1`

##### Enterprise

- 1.10.x+ent - 1.15.11+ent
- 1.16.5+ent
- 1.17.1+ent

#### Issue

> When a Vault TCP listener is configured with the `deny_unauthorized` [proxy_protocol_behavior](/vault/docs/configuration/listener/tcp#proxy_protocol_behavior)
> and receives a request from an untrusted upstream connection, Vault will no longer
> respond to requests via that listener.

Reviewer 2: ```suggestion
Vault TCP listeners configured with the `deny_unauthorized` [proxy_protocol_behavior](/vault/docs/configuration/listener/tcp#proxy_protocol_behavior) process, but do not respond to, requests from untrusted upstream connections,
```
Style correction: write in active voice

> Vault TCP listeners configured with the `deny_unauthorized` [proxy_protocol_behavior](/vault/docs/configuration/listener/tcp#proxy_protocol_behavior)
> which receive a request from an untrusted upstream connection will close, causing
> Vault to no longer respond to requests via that listener.

Reviewer 2: ```suggestion
Vault TCP listeners configured with the `deny_unauthorized` [proxy_protocol_behavior](/vault/docs/configuration/listener/tcp#proxy_protocol_behavior)
close if they receive a request from an untrusted upstream connection. As a result,
Vault no longer responds to *any* request received through that listener.
```

Style correction: write in active voice

> This leaves the Vault API vulnerable to a denial-of-service (DoS) attack by requests

Reviewer 2: What is "this"? The closed listener or the fact that Vault can't receive requests through it?

> Being able to close listeners in this way leaves the Vault API vulnerable to a
> denial-of-service (DoS) attack by requests which are intentionally made by untrusted
> upstream connections, causing it to become unavailable on that node.

Reviewer 2: ```suggestion
Being able to force-close listeners with intentionally untrusted connections
leaves the Vault API vulnerable to a denial-of-service (DoS) attacks, which
could cause the API to become unavailable on that node.
```

> This issue is addressed in Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent and later.

Reviewer 2: What is "this"? The fact that the listener closes or that it leaves Vault susceptible to DoS attacks?

Author: Both,  by stopping the listener from closing we stop the ability for the DoS to be exploited in this way.

> This could lead to a situation where the Vault API could be vulnerable to a denial-of-service
> (DoS) attack, if requests are intentionally made by untrusted upstream connections
> to the Vault API, causing it to become unavailable on that node.
> 
> This issue is addressed in Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent and later.

Reviewer 2: ```suggestion
The fact that the listener takes the time to process the request before ignoring it, leaves the listener vulnerable to denial-of-service (DoS) attacks by requests intentionally sent from untrusted upstream connections.

We recommend upgrading to Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent or later to address the vulnerability.
```

Style correction: write in active voice

Author: > The fact that the listener takes the time to process the request before ignoring it, leaves the listener 

The vulnerability is such that you can prevent Vault from serving its API entirely - therefore no further requests can be successful to that Vault node. 

It's not that the listener takes a long time to process anything, it doesn't it, it's quick to process but closes when we stop serving HTTP in response to the way the library returns the error from the listener's `Accept` method.

Author: > We recommend upgrading to Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent or later to address the vulnerability.

These versions aren't out yet, so I was just following the pattern we've used in previous known issues to call out the versions that (will) fix it.

> This issue is addressed in Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent and later.

Reviewer 2: ```suggestion
The vulnerability is addressed in Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent and later.
```
